src/sites/twitch-twilight/modules/chat/input.jsx

```diff
@@ -647,21 +647,49 @@ export default class Input extends Module {
 
 
 	// eslint-disable-next-line class-methods-use-this
-	sortFavorites(results) {
-		if (!this.chat.context.get('chat.tab-complete.prioritize-favorites')) {
-			return results;
+	sortEmotes(emotes, input) {
+		const preferFavorites = this.chat.context.get('chat.tab-complete.prioritize-favorites');
+		const lowerCaseInput = input.toLowerCase();
+
+		const locale = new Intl.Collator();
+		const localeCI = new Intl.Collator(undefined, {sensitivity: 'accent'}); // case insensitive
+
+		const startsWithInput = new Set();
```

Comment:
Constructing extra data structures is unnecessary. The objects in the variable you renamed to `emotes` are not persistent. You can safely extend them with variables indicating how the emote matches the string.

To that note, I would suggest that instead of doing this here at all, that `doesEmoteMatchTerm` is extended to return an integer depending on how the emote name matched, and that the integer is stored on emote result objects, that way we do less string checking.

Reply:
So, my suggestion is:

* Modify `doesEmoteMatchTerm` to return a number. 0 if there's no match so false-y checks can still be done.
* Modify `getTwitchEmoteSuggestions` and `getEmoteSuggestions` to store the resulting number on result objects.
* Use that number in the `sortEmotes` method rather than doing any further string checks against the input.

Reply:
Addressed in f85f0c4.